### PR TITLE
Adds metrics for failed communications with keywhiz server.

### PR DESCRIPTION
**Pull request by stfinney**
Relies on a change to SquareMetrics: https://github.com/square/go-sq-metrics/pull/6

r: @mcpherrinm @csstaub @worldwise001

### CODE REVIEW

**Comment by csstaub** [comment]
You gotta bump the vendored dependency to the updated version.

File: client.go [hunk 1]
```diff
@@ -189,18 +210,24 @@ func (c Client) RawSecretList() (data []byte, ok bool) {
...
 		return nil, false
 	}
+	c.failCountClear()
+	c.lastSuccessUpdate()
 	return data, true
 }
 
 // SecretList returns a slice of unmarshalled Secret structs after requesting a listing of secrets.
+// only used by tests
+// TODO: do we need this?
```
Comment (csstaub) [inline client.go]: Make this TODO into an issue

File: client.go [hunk 2]
```diff
@@ -65,9 +68,19 @@ func (e SecretDeleted) Error() string {
 	return "deleted"
 }
 
+func (c Client) failCountInc() {
+	metrics.GetOrRegisterCounter("runtime.server.fails", c.metricsHandle.Registry).Inc(1)
+}
+func (c Client) failCountClear() {
+	metrics.GetOrRegisterCounter("runtime.server.fails", c.metricsHandle.Registry).Clear()
```
Comment (csstaub) [inline client.go]: Instead of calling this over and over again, just call it once and then reuse it?

File: client.go [hunk 3]
```diff
@@ -141,23 +155,29 @@ func (c Client) RawSecret(name string) (data []byte, err error) {
...
 	if err != nil {
 		c.Errorf("Error reading response body for secret %v: %v", name, err)
+		c.failCountInc()
 		return nil, err
 	}
 
 	switch resp.StatusCode {
 	case 200:
+		c.failCountClear()
+		c.lastSuccessUpdate()
```
Comment (mcpherrinm) [inline client.go]: We always call these two together.  Maybe they should be one function.  c.MarkSuccess() or something.

**Comment by csstaub** [comment]
LGTM

File: client.go [hunk 4]
```diff
@@ -46,9 +48,12 @@ var ciphers = []uint16{
 // Client basic struct.
 type Client struct {
 	*klog.Logger
-	http   func() *http.Client
-	url    *url.URL
-	params httpClientParams
+	http          func() *http.Client
+	url           *url.URL
+	params        httpClientParams
+	metricsHandle *sqmetrics.SquareMetrics
```
Comment (mcpherrinm) [inline client.go]: I think this is unused -- we just use the two entries below

**Comment by mcpherrinm** [comment]
LGTM

**Pull request merged**